src/js/controller/model.js

```diff
@@ -94,11 +94,24 @@ define([
                     this.set('buffer', data.bufferPercent);
                 /* falls through */
                 case events.JWPLAYER_MEDIA_META:
+                    var metaStreamType = utils.streamType(data.duration, data.seekableRange, this.get('minDvrWindow'));
+                    this.set('streamType', metaStreamType);
+                    this.set('seekableRange', data.seekableRange);
```

Comment:
Do html5 and flash send `seekableRange`? This should require changes to those providers and testing of HLS streams in Safari and in a browser with Flash and primary="flash".